service/dap/server.go

```diff
@@ -786,6 +807,13 @@ func (s *Session) handleRequest(request dap.Message) {
 }
 
 func (s *Session) send(message dap.Message) {
+	if event, ok := message.(*dap.OutputEvent); ok {
```

Comment:
You can't hijack the send method to do other things, it should be done by the caller of this method, or maybe in a method that both waits for the readers to finish and then sends the terminated event.